modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -633,4 +644,38 @@ public static boolean isParameterContainer(Set<Class<? extends Annotation>> anno
                 .filter(field -> isMultiLevelMetadataKeyId(field, field.getType(), typeLoader))
                 .collect(new ImmutableListCollector<>());
     }
+
+    public static String getComponentModelTypeName(Object component)
+    {
+        if (component instanceof OperationModel)
+        {
+            return OPERATION;
+        }
+        else if (component instanceof ConfigurationModel)
+        {
+            return CONFIGURATION;
+        }
+        else if (component instanceof ConnectionProviderModel)
+        {
+            return CONNECTION_PROVIDER;
+        }
+        else if (component instanceof SourceModel)
+        {
+            return SOURCE;
+        }
+
+        throw new IllegalArgumentException(format("Component '%s' is not an instance of any known model type [%s, %s, %s, %s]",
+                                                  component.toString(),
+                                                  CONFIGURATION, CONNECTION_PROVIDER, OPERATION, SOURCE));
+    }
+
+    public static String getModelName(Object model)
```

Comment:
I have doubts about this method, but It happens that sometimes a method receives an EnrichableModel, and even though I know for sure that the received model implements Named, there is no common interface for Operation, Config, Provider and  Source that shares being Enrichable and Named. Long story short, I can either do this or received the modelName as an argument of the method.